src/evalfunc.c

```diff
@@ -2152,7 +2150,7 @@ static funcentry_T global_functions[] =
 			ret_string,	    f_inputsecret},
     {"insert",		2, 3, FEARG_1,	    arg23_insert,
 			ret_first_arg,	    f_insert},
-    {"instanceof",	2, 2, FEARG_1,	    arg2_instanceof,
+    {"instanceof",	2, 19, FEARG_1,	    arg219_instanceof,
```

Comment:
Instead of using the maximum number of arguments (19) to specify the variable length arguments, can you use -1?  You can then check for -1 in the internal_func_check_arg_types() function and skip the compile time checks.  You need to use runtime checks in the instanceof() function.

Reply:
> Instead of using the maximum number of arguments (19) to specify the variable length arguments, can you use -1? You can then check for -1 in the internal_func_check_arg_types() function

OK, sounds better; I'll take a look at that.

> and skip the compile time checks.

I'll try to keep a compile time check if it's not too much trouble. `instanceof()` has simple, specific and limited argument requirements. `instanceof()` is a strange case. Strictly, it should be part of the language; for example Java does
```
if (<var> instanceof <type>)
```
Deferring the check to runtime seems wrong.

> You need to use runtime checks in the instanceof() function.

There is a runtime check in f_instanceof; or am I missing something?

Reply:
Yes.  The runtime check is already present.  You can check the first two argument types at compile time (minimum number of arguments).  The rest is anyway checked at runtime.